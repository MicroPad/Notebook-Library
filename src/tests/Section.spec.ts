import { Note, Section } from '../index';
import { TestUtils } from './TestUtils';

describe('Section', () => {
	it('should construct', () => {
		// Arrange
		const title = 'test';

		// Act
		const res = TestUtils.makeSection(title);

		// Assert
		expect(res).toBeInstanceOf(Section);
		expect(res.title).toEqual(title);
	});

	describe('addSection', () => {
		let parent: Section;
		let child: Section;

		beforeEach(() => {
			parent = TestUtils.makeSection('test parent');
			child = TestUtils.makeSection('test child');
		});

		it('should add a new section', () => {
			//Arrange
			// Act
			const res = parent.addSection(child);

			// Assert
			expect(res.sections[0]).toEqual(child);
		});

		it('should create a new object', () => {
			//Arrange
			// Act
			const res = parent.addSection(child);

			// Assert
			expect(res).not.toBe(parent);
		});
	});

	describe('addNote', () => {
		let parent: Section;
		let child: Note;

		beforeEach(() => {
			parent = TestUtils.makeSection('test parent');
			child = TestUtils.makeNote('test child');
		});

		it('should add a new note', () => {
			//Arrange
			// Act
			const res = parent.addNote(child);

			// Assert
			expect(res.notes[0]).toEqual(child);
		});

		it('should create a new object', () => {
			//Arrange
			// Act
			const res = parent.addNote(child);

			// Assert
			expect(res).not.toBe(parent);
		});
	});

	it('should generate XML Object with required data', () => {
		// Arrange
		let section = TestUtils.makeSection('test parent');
		let child = TestUtils.makeSection('test child');
		child = child.addNote(TestUtils.makeNote('test note'));

		section = section.addSection(child);

		// Act
		const res = section.toXmlObject();

		// Assert
		expect(res).toMatchSnapshot();
	});
});
