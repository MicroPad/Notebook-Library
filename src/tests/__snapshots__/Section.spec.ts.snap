// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should generate XML Object with required data 1`] = `
Object {
  "section": Object {
    "$": Object {
      "title": "test parent",
    },
    "note": Array [],
    "section": Array [
      Object {
        "$": Object {
          "title": "test child",
        },
        "note": Array [
          Object {
            "$": Object {
              "time": "1970-01-01T12:00:00.001+12:00",
              "title": "test note",
            },
            "addons": Array [],
            "bibliography": Array [],
          },
        ],
        "section": Array [],
      },
    ],
  },
}
`;
